### PR TITLE
examples: add gradient-boosting-regression

### PR DESCRIPTION
**Pull request by Heluwe**
Add sklearn gradient boosting regression example, see https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_regression.html?highlight=feature_importance

### CODE REVIEW

**Comment by yorkie** [comment]
Thank you @Heluwe, there are some style nits reported by GitHub, could you please fix them :)
